src/metabase/driver/mysql.clj

```diff
@@ -332,7 +342,7 @@
                        (set/rename-keys {:dbname :db})
                        (dissoc :ssl))]
        (-> (dbspec/mysql details)
-           (sql-jdbc.common/handle-additional-options details))))))
+           (sql-jdbc.common/handle-additional-options (add-program-name-option details)))))))
```

Comment:
```suggestion
           (sql-jdbc.common/handle-additional-options details)
           (assoc :connectionAttributes (str "program_name:" (str/replace config/mb-version-and-process-identifier "," "_"))))))))
```

Rather than messing with "additional options", you can just add it to the spec which will end up in JDBC properties and the driver will pick it up.

The above suggestion is just a simple example of how to make it work without changing additional options. Probably you'd still want the logic to be in `add-program-name-option`.

Reply:
I don't fully understand the code, but it would be very helpful if we allow additional options to override our hardcoded stuff. We've had several issues, where we hardcode something, which means it cannot be overridden.

Reply:
@pawit-metabase , done.  That is a little cleaner.

@flamber , this is just the string that basically shows up in the DB if you query for active connections.  See the 2nd comment above ^.  I am open to allowing override of this string, for very sophisticated users (most likely), but this particular PR is doing nothing more than extending the work already done for Oracle and Postgres, to MySQL (see #15679).  In any event, this should be a vast improvement over seeing either an empty string, or some generic name like `connector/j` or whatever.